pkg/controllers/report/aggregate/controller.go

```diff
@@ -530,10 +530,16 @@ func (c *controller) backReconcile(ctx context.Context, logger logr.Logger, _, n
 	for _, result := range merged {
 		results = append(results, result)
 	}
+	// Delete report if no results (fix for orphaned PolicyReports)
 	if len(results) == 0 {
 		if report != nil {
 			return deleteReport(ctx, report, c.client)
 		}
+		// Delete any existing PolicyReports for this resource
+		_ = c.client.Wgpolicyk8sV1alpha2().PolicyReports(namespace).Delete(ctx, name, metav1.DeleteOptions{})
```

Comment:
as I checked the code the line above

```go
return deleteReport(ctx, report, c.client)
```

should already delete empty reports. Do you see any logs that indicates that the deletion do not work or is not executed probably?